src/request/components/RequestDetailPanelRequestMiddleware.tsx

```diff
@@ -0,0 +1,107 @@
+import { trainCase } from '../../lib/StringUtilities';
+
+import * as _ from 'lodash';
+import * as React from 'react';
+
+interface IFlattenedMiddleware {
+    depth: number;
+    middleware: {
+        name: string,
+        packageName: string,
+        headers: { [key: string]: { value: string, isCurrent: boolean } }
+    };
+}
+
+export interface IRequestMiddlewareProps {
+    middleware: IFlattenedMiddleware[];
+}
+
+export class RequestMiddleware extends React.Component<IRequestMiddlewareProps, {}> {
+    public render() {
+        return (
+            <div className='tab-request-middleware'>
+                <div className='tab-request-title'>Middleware</div>
+                <br />
+                <table className='table table-bordered table-striped table-selectable tab-request-middleware-table'>
+                    <thead>
+                        <tr className='table-col-title-group'>
+                            <th width='10%'><span className='table-col-title'>Ordinal</span></th>
+                            <th width='25%'><span className='table-col-title'>Name</span></th>
+                            <th width='20%'><span className='table-col-title'>Type</span></th>
+                            <th width='10%'><span className='table-col-title'>Modify</span></th>
+                            <th width='25%'><span className='table-col-title'>Parameter</span></th>
+                            <th />
+                        </tr>
+                    </thead>
+                    <tbody>
+                        { this.props.middleware.map((middlewareRow, index) => this.renderMiddlewareRow(index + 1, middlewareRow)) }
+                    </tbody>
+                </table>
+            </div>
+        );
+    }
+
+    private renderMiddlewareRow(ordinal: number, middleware: IFlattenedMiddleware) {
+        return (
+            <tr>
+                <td>{ordinal}</td>
+                <td>{this.renderName(middleware.middleware.name, middleware.depth)}</td>
+                <td>{middleware.middleware.packageName}</td>
+                <td>{_.size(middleware.middleware.headers) > 0 ? 'Header' : '-'}</td>
```

Comment:
Thinking... this is something that would need to change if/when we track more than headers.